### PR TITLE
[MU4] Fix musescore#12028: Fix messaging when opening non-existing scores

### PR DESCRIPTION
**Pull request by iwoithe**
Resolves: #12028

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [ ] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by cbjeukendrup** [comment]
Rebase needed :)

**Comment by iwoithe** [comment]
Done. :slightly_smiling_face:

File: src/engraving/infrastructure/mscreader.cpp [hunk 1]
```diff
@@ -26,6 +26,8 @@
 #include "io/dir.h"
 #include "serialization/zipreader.h"
 #include "serialization/xmlstreamreader.h"
+#include "engraving/engravingerrors.h"
+#include "libmscore/masterscore.h"
```
Comment (igorkorsukov) [inline src/engraving/infrastructure/mscreader.cpp:30]: This inclusion is not required.

**Review by igorkorsukov — APPROVED** [review]

**Comment by cbjeukendrup** [comment]
@bkunda Actually that would defeat the purpose of this PR: the point of this PR is to be a bit more specific than _"an error had occurred"_. Would it be an idea to maintain the specific-ness of this PR? So, if a file cannot be opened...
- ... if it's because of corruption, show the corruption dialog
- ... otherwise, show that simple dialog you propose, but with the more specific explanation implemented by this PR.
Of course, the exact wordings of these explanations can be discussed.

**Comment by bkunda** [comment]
@cbjeukendrup can you please follow up on this one? Many thanks!

**Comment by cbjeukendrup** [comment]
@iwoithe I created a follow-up PR with some additional changes here: https://github.com/musescore/MuseScore/pull/18376
That PR contains your PR, so I will close this one, but you can consider it merged. Thanks for the contribution!

**Pull request closed without merging**